src/renderer/dx/DxRenderer.cpp

```diff
@@ -112,6 +283,134 @@ DxEngine::~DxEngine()
     return S_OK;
 }
 
+int DbgPrint(const char* format, ...)
+{
+    static char s_printf_buf[1024];
+    va_list args;
+    va_start(args, format);
+    _vsnprintf_s(s_printf_buf, sizeof(s_printf_buf), format, args);
+    va_end(args);
+    OutputDebugStringA(s_printf_buf);
+    return 0;
+}
+
+Microsoft::WRL::ComPtr<ID3DBlob>
+_CompileShader(

```

Comment:
Also probably needs a doc comment, you can see the format we usually use on other methods in this file